### ISSUE TITLE
Support role_arn and source_profile when reading aws credentials

### ISSUE THREAD

**Issue by alzadude** [cognitect-labs/aws-api#172]
Hi there,

It seems the library doesn't yet support role-based access, it only supports `aws_access_key_id` and `aws_secret_access_key` unless I missed something?

Supporting `role_arn` and `source_profile` would allow users to connect using different roles into different aws accounts etc etc.

Thanks 🙂

**Comment by ian-moore** [comment]
There is a `profile-credentials-provider` which uses a profile name. https://github.com/cognitect-labs/aws-api/blob/d85b0078bdf4cf700ecfe0828bc04d9d81995ab6/src/cognitect/aws/credentials.clj#L210

**Comment by scarytom** [comment]
I agree with @alzadude, this seems unsupported.  When calling `profile-credentials-provider` and specifying the name of a profile that is defined with a source_profile like `bar` is below then it fails.

sample credentials file:
```[foo]
aws_access_key_id=AWDFASDFASEFAWEFWEF
aws_secret_access_key=324er923r890seufosiufw489ufoufresoifus

[bar]
role_arn = arn:aws:iam::123456789012:role/Administrator
source_profile = foo
```

**Comment by MiyamotoAkira** [comment]
I needed this to work, so I wrote some code  using the STS example provided by Cognitect, and butchering a copy of the profile-credentials-provider as a first dirty version:
[gist client.clj](https://gist.github.com/MiyamotoAkira/898332885d613124534fc6a188ec88cf)

I will improve that for my use, but in case someone needs to do something similar.
